.github/workflows/ci.yml

```diff
@@ -659,7 +669,7 @@ jobs:
 
       # XPMEM
       - name: Checkout XPMEM
-        uses: actions/checkout@v2
+        uses: actions/checkout@v4
```

Comment:
Why `v4` not `v2`?

Reply:
We tentatively decided to update because it seems there have been bug fixes in the checkout actions since `v2` that could affect submodules:

https://github.com/actions/checkout/blob/main/CHANGELOG.md

But I'm not sure we really _need_ to update... but it's working so I don't see the harm.

Does that answer the question, or do you have a concern about updating @wrrobin?